operator/pkg/helmreconciler/apply.go

```diff
@@ -174,27 +173,22 @@ func (h *HelmReconciler) ApplyObject(obj *unstructured.Unstructured) error {
 		return nil
 	}
 
-	backoff := wait.Backoff{Duration: time.Millisecond * 10, Factor: 2, Steps: 3}
-	return retry.RetryOnConflict(backoff, func() error {
-		err := h.client.Get(context.TODO(), objectKey, receiver)
-		switch {
-		case errors2.IsNotFound(err):
-			scope.Infof("creating resource: %s", objectStr)
-			err = h.client.Create(context.TODO(), obj)
-			if err != nil {
-				return fmt.Errorf("failed to create %q: %w", objectStr, err)
-			}
-			return nil
-		case err == nil:
-			scope.Infof("updating resource: %s", objectStr)
-			// The correct way to do this is with a server-side apply. However, this requires users to be running Kube 1.16.
-			// When we no longer support < 1.16 use the code described in the linked issue.
-			// https://github.com/kubernetes-sigs/controller-runtime/issues/347
-			if err := applyOverlay(receiver, obj); err != nil {
-				return err
-			}
-			return h.client.Update(context.TODO(), receiver)
+	err := h.client.Get(context.TODO(), objectKey, receiver)
+	switch {
+	case errors2.IsNotFound(err):
+		scope.Infof("creating resource: %s", objectStr)
+		if err := h.client.Create(context.TODO(), obj); err != nil {
+			return fmt.Errorf("failed to create %q: %w", objectStr, err)
 		}
 		return nil
-	})
+	case err == nil:
+		scope.Infof("updating resource: %s", objectStr)
+		// use server-side apply, require kubernetes 1.16+
```

Comment:
I am not sure we are ready to drop Kubernetes 1.15 support

Reply:
yep that's a concern, so I am not sure whether we need to get this in for 1.8, probably better to wait for 1.9 or later.

Reply:
we should preserve the old way alongside - so what we need is a selection based on the k8s version somewhere and select the appropriate method based on that.

Reply:
done